context/app/routes_browse.py

```diff
@@ -113,8 +114,24 @@ def details_vitessce(type, uuid):
     parent_uuid = request.args.get('parent') or None
     marker = request.args.get('marker') or None
     parent = client.get_entity(parent_uuid) if parent_uuid else None
+    epic_uuid = None
+    epic_entity = None
+    if 'segmentation_mask' in entity.get('vitessce-hints'):
+        epic_entity = entity
+        epic_uuid = uuid
+        if parent is None:
+            ancestors = entity.get('immediate_ancestors')
+            if len(ancestors) > 0 and ancestors[0]['entity_type'] == 'Dataset':
```

Comment:
It would be preferable to use `immediate_ancestors_ids` and make a request to get the ancestor entities. `immediate_ancestors` is somewhat deprecated as it greatly increases the ES document size. We could have the query only return datasets.